linera-views/src/register_view.rs

```diff
@@ -58,29 +60,42 @@ where
     }
 
     fn rollback(&mut self) {
+        self.is_default = false;
         self.update = None;
         *self.hash.get_mut() = self.stored_hash;
     }
 
     fn flush(&mut self, batch: &mut Batch) -> Result<(), ViewError> {
-        if let Some(value) = self.update.take() {
-            let key = self.context.base_tag(KeyTag::Value as u8);
-            batch.put_key_value(key, &value)?;
-            self.stored_value = value;
+        if self.is_default {
+            batch.delete_key_prefix(self.context.base_key());
+        }
+        // If there is clearing then we want to write the value only if it has been set afterward
+        if !self.is_default {
+            if let Some(value) = self.update.take() {
+                let key = self.context.base_tag(KeyTag::Value as u8);
+                batch.put_key_value(key, &value)?;
+                self.stored_value = value;
+            }
         }
         let hash = *self.hash.get_mut();
-        if self.stored_hash != hash {
+        // In the scenario where we do a clear
+        // and stored_hash = hash, we need to update the
+        // hash, otherwise, we will recompute it while this
+        // can be avoided.
+        if self.stored_hash != hash || self.is_default {
```

Comment:
I'd find it easier to read if we just set `self.stored_hash = None` in the `is_default` case above, right after `batch.delete_key_prefix(self.context.base_key());`. Then we wouldn't need the `is_default` case, I think?